### PR TITLE
feat: support match mappings

### PR DESCRIPTION
**Pull request by yehuohan**
Add `match_mappings` supports. The detailed description can see https://github.com/phaazon/hop.nvim/pull/235.

`match_mappings` is mainly to support jump to non-english text which may be multi-byte characters, so this PR supports multi-byte characters for hop vertical too (seemed also fix hop vertical crash issue when win_view.leftcol > 0).

### CODE REVIEW

**Comment by smoka7** [comment]
can you also add tests for this

File: lua/hop/jump_target.lua [hunk 1]
```diff
@@ -68,32 +101,35 @@ local function mark_jump_targets_line(ctx)
...
+    shifted_line = shifted_line:sub(1, ctx.cursor_pos[2] - col_bias + 1)
+  end
+  if shifted_line == '' and ctx.col_offset > 0 then
+    -- No possible position to place target
+    return jump_targets
   end
 
+  -- Match context for regex match
+  local match_context = {
+    cursor_fcol = ctx.cursor_pos.fcol,
```
Comment (smoka7) [inline lua/hop/jump_target.lua:125]: add the type for this field

File: lua/hop/mappings.lua [hunk 2]
```diff
@@ -0,0 +1,28 @@
+local M = {}
+
+-- Checkout mappings with the chars of pattern as key
+function M.checkout(pat, opts)
```
Comment (smoka7) [inline lua/hop/mappings.lua:4]: add types

File: lua/hop/jump_target.lua (hunk 1 above)
Comment (yehuohan) [reply lua/hop/jump_target.lua:129]: added.

File: lua/hop/mappings.lua (hunk 2 above)
Comment (yehuohan) [reply lua/hop/mappings.lua:6]: added.

**Comment by yehuohan** [comment]
> can you also add tests for this

I'll try to add testcases for match-mappings from studying hop_spec.lua :)

**Comment by yehuohan** [comment]
@smoka7 I have added tests for match mappings. Can you make a review on this PR?

**Comment by smoka7** [comment]
It looks good, I wanna add some mappings and make some small changes and then merge it.
Thank you for your work.

**Comment by smoka7** [comment]
hey I want to push here but i get this error `remote: Permission to yehuohan/hop.nvim.git denied to smoka7.` any idea what's causing this?

**Comment by yehuohan** [comment]
> hey I want to push here but i get this error `remote: Permission to yehuohan/hop.nvim.git denied to smoka7.` any idea what's causing this?

Sorry, I'm not sure what's happend about this. My fork `hop.nvim` is public and you should have the access to it (The PR https://github.com/smoka7/hop.nvim/pull/5 you had merged can confirm this).
It there any information about my fork `hop.vim` I can provide for you?

**Comment by smoka7** [comment]
It's probably because you forked from phazon, I'm not the in the maintainer list.
Could you add me to the [collaborators](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-access-to-your-personal-repositories/inviting-collaborators-to-a-personal-repository) to your fork.

**Comment by yehuohan** [comment]
@smoka7 I added. Can you try again?

**Comment by smoka7** [comment]
Thanks, it's fixed. See if the changes are okay with you, Then I'll merge it.

**Comment by yehuohan** [comment]
Great! It's more readable.

**Pull request merged**